hawkbit-repository/hawkbit-repository-jpa/src/test/java/org/eclipse/hawkbit/repository/jpa/ActionStatusUpdateHandlerServiceTest.java

```diff
@@ -0,0 +1,119 @@
+public class ActionStatusUpdateHandlerServiceTest extends AbstractJpaIntegrationTest {
+
+    @Test
+    @Description("Verifies that the status update(finished state) for a distribution id is updated in the action database.")
+    public void verifyStatusUpdate() {
+
+        String controllerId = "com.bosch.iot.dm.ac:cac";
+        String tenant = "test";
+
+        // generate data in database
+        JpaSoftwareModule swModule = this.generateSoftwareModule(tenant);
+        JpaDistributionSet ds = generateDistributionSet(swModule, tenant);
+        JpaTarget target = generateSampleTarget(1L, controllerId, tenant);
+        JpaAction action = generateAction(101L, ds, target, tenant);
+
+        ActionStatusUpdateHandlerService rolloutStatusHandlerService = new ActionStatusUpdateHandlerService(
+                this.controllerManagement, this.entityFactory, this.systemSecurityContext);
+
+        // initiate the test
+        ActionStatusUpdateEvent targetStatus = new ActionStatusUpdateEvent("default", action.getId(), Status.FINISHED,
+                new ArrayList<>());
+        rolloutStatusHandlerService.handle(targetStatus);
+
+        // verify if intended dataSetId is really installed
+        Long installedId = this.targetRepository.findById(target.getId()).get().getInstalledDistributionSet().getId();
+        assertThat(installedId).isEqualTo(ds.getId()).as("last installedId must be updated in the database");
+
+        // verify that action is database is marked inactive.
+        Optional<JpaAction> activeAction = this.actionRepository.findById(action.getId());
+        assertThat(activeAction.isPresent()).isTrue().as("action must be present in the database after status update");
+        assertThat(activeAction.get().isActive()).isFalse().as("on finished status update, the action must be marked in active");
+        assertThat(activeAction.get().getStatus()).isEqualTo(Status.FINISHED).as("saved action must have finished status");
+        
+        // clean up data - start
+        this.actionRepository.deleteById(action.getId());
+        this.softwareModuleRepository.deleteById(swModule.getId());
+        this.targetRepository.deleteById(target.getId());
+        this.distributionSetRepository.deleteById(ds.getId());
+    }
+
+    private JpaDistributionSet generateDistributionSet(final JpaSoftwareModule swModule, final String tenant) {
+        JpaDistributionSet ds = new JpaDistributionSet();
+        ds.setName("test ds");
+        ds.setVersion("v1");
+        JpaDistributionSetType type = this.distributionSetTypeRepository.findById(4L).get();
+        ds.setType(type);
+        ds.setTenant(tenant);
+        ds.addModule(swModule);
+        ds.setRequiredMigrationStep(false);
+        return this.distributionSetRepository.save(ds);
+    }
+
+    private JpaSoftwareModule generateSoftwareModule(final String tenant) {
+        JpaSoftwareModuleType type = this.softwareModuleTypeRepository.findById(3L).get();
```

Comment:
Don't relay on hard coded entity IDs within our test. This is the reason why your CircleCI build is failing => the `Optional.get()` is not present

Reply:
From offline discussion:
- It was recommended to use 
 - Ex: testdataFactory.createSoftwareModule(TestdataFactory.SM_TYPE_APP) creates a software module.